src/components/ChannelList/ChannelList.js

```diff
@@ -1,543 +1,361 @@
-  }
-
-  // eslint-disable-next-line sonarjs/cognitive-complexity
-  handleEvent = async (e) => {
-    if (e.type === 'user.presence.changed') {
-      let newChannels = this.state.channels;
-
-      newChannels = newChannels.map((channel) => {
-        if (!channel.state.members[e.user.id]) return channel;
-
-        channel.state.members.setIn([e.user.id, 'user'], e.user);
-
-        return channel;
-      });
-
-      this.setState({ channels: [...newChannels] });
-    }
-
-    if (e.type === 'message.new') {
-      !this.props.lockChannelOrder && this.moveChannelUp(e.cid);
-    }
-
-    // make sure to re-render the channel list after connection is recovered
-    if (e.type === 'connection.recovered') {
-      this.setState((prevState) => ({
-        connectionRecoveredCount: prevState.connectionRecoveredCount + 1,
-      }));
-    }
-
-    // move channel to start
-    if (e.type === 'notification.message_new') {
-      // if new message, put move channel up
-      // get channel if not in state currently
-      if (
-        this.props.onMessageNew &&
-        typeof this.props.onMessageNew === 'function'
-      ) {
-        this.props.onMessageNew(this, e);
-      } else {
-        const channel = await this.getChannel(e.channel.type, e.channel.id);
-        // move channel to starting position
-        this.setState((prevState) => ({
-          channels: uniqBy([channel, ...prevState.channels], 'cid'),
-        }));
-      }
-    }
-
-    // add to channel
-    if (e.type === 'notification.added_to_channel') {
-      if (
-        this.props.onAddedToChannel &&
-        typeof this.props.onAddedToChannel === 'function'
-      ) {
-        this.props.onAddedToChannel(this, e);
-      } else {
-        const channel = await this.getChannel(e.channel.type, e.channel.id);
-        this.setState((prevState) => ({
-          channels: uniqBy([channel, ...prevState.channels], 'cid'),
-        }));
-      }
-    }
+  // Set a channel with id {customActiveChannel} as active and move it to the top of the list.
+  // Of customActiveChannel prop is absent, then set the first channel in list as active channel.
+  const activeChannelHandler = (channels) => {
+    const {
+      setActiveChannel,
+      setActiveChannelOnMount,
+      customActiveChannel,
+      watchers,
+      options,
+    } = props;
 
-    // remove from channel
-    if (e.type === 'notification.removed_from_channel') {
-      if (
-        this.props.onRemovedFromChannel &&
-        typeof this.props.onRemovedFromChannel === 'function'
-      ) {
-        this.props.onRemovedFromChannel(this, e);
-      } else {
-        this.setState((prevState) => {
-          const channels = prevState.channels.filter(
-            (channel) => channel.cid !== e.channel.cid,
-          );
-          return {
-            channels,
-          };
-        });
-      }
+    if (
+      !channels ||
+      channels.length === 0 ||
+      channels.length > (options.limit || MAX_QUERY_CHANNELS_LIMIT)
```

Comment:
Why do we need this condition here?

Reply:
Its to ensure we execute this only on first page load. But maybe makes sense to move this logic to hook (usePaginatedChannels) itself

Reply:
Got it, thanks. And I agree with you about moving it to the paginated channels hook 👍